src/osd/ECBackend.cc

```diff
@@ -500,6 +500,8 @@ void ECBackend::continue_recovery_op(
 	assert(!op.recovery_progress.first);
 	dout(10) << __func__ << ": canceling recovery op for obj " << op.hoid
 		 << dendl;
+	// XXX: Doing this on the read error handling just loops in recovery
+	// How would this case ever actually happen?
```

Comment:
You can't just use cancel_pull() in the read error handling because you know something the parent doesn't: that some of the sources are faulty.  Here, I think we are failing because a source went down (not in the up or acting set, or it would have triggered peering to restart) and we no longer have enough copies.  I think it's ok to just cancel the recovery op here because the parent would already have updated its missing_loc etc when it got the log update.  Either the object later becomes recoverable again and recovery will retry or not.